### ISSUE TITLE
SyntaxError: Identifier 'frames' has already been declared

### ISSUE THREAD

**Issue by ato** [webrecorder/wombat#72]
The preamble added by Wombat.wrapScriptTextJsProxy to every JavaScript file declares the following locals:

```js
let window = _____WB$wombat$assign$function_____("window");
let self = _____WB$wombat$assign$function_____("self");
let document = _____WB$wombat$assign$function_____("document");
let location = _____WB$wombat$assign$function_____("location");
let top = _____WB$wombat$assign$function_____("top");
let parent = _____WB$wombat$assign$function_____("parent");
let frames = _____WB$wombat$assign$function_____("frames");
let opener = _____WB$wombat$assign$function_____("opener");
```

If the script itself uses any of those names as locals then it causes an error:

    SyntaxError: Identifier 'frames' has already been declared

Discovered in the wild on the wild on a pywb replay of https://gladybird.com/ in which [game.js](https://gladybird.com/game.js) tries to declare a frame counter with:

```js
let frames = 0;
```

**Comment by ato** [comment]
We're trying a workaround <https://github.com/nla/pywb/commit/fe97e7e8db092810e9e6beb417067fe476c8c9d2> for this by wrapping the script in additional block after wombat's locals are declared. Effectively:

```js
let frames = _____WB$wombat$assign$function_____("frames");
{
   let frames = 0;
   // .. the rest of the original script
}
```

This solves it for the particular case we had reported, although this is far from a perfect solution as it'll still throw SyntaxError if the script uses `var frames` instead of `let frames`. I also worried it would also have the side effect of breaking code like this:
    

**test.js**
```js
let x = "Hello world!"
```
    
**test.html**
```html
<script src="test.js"></script>
<script>
document.write(x);
</script>
```

However it turns out pywb is already wrapping rewritten scripts in a block so I guess the workaround doesn't make that any worse than it already is.

**Comment by ikreymer** [comment]
That's a pretty good workaround.. yes, there is an issue with globals being wrapped. Luckily, this type of use with globals is not as is not as common, and less so with modern websites.

As far as `var`, it appears that `var` with `self`, `parent, `opener`, `frames`, is allowed but not with `window`, `location`, `document` and `top`.

Perhaps it could be worth adding a check for, say `var frames` or `var opener` and skipping if found. Then again, it'll need to be a regex and could slow down processing, so perhaps not needed until it is found 'in the wild'.

(I guess this issue is shared between wombat/pywb/wabac.js since this wrapping happens in all 3 places currently).

**Comment by ikreymer** [comment]
Mostly fixed in 3.3.0 release by implementing the workaround suggested by @ato in https://github.com/webrecorder/wombat/issues/72#issuecomment-886342725

Other name collisions still possible in theory, but current issue is addressed.
